### PR TITLE
Save settings immediately upon changing in RunActivity.exe

### PR DESCRIPTION
**Pull request by YoRyan**
Changes the code to save whitelisted settings immediately upon change. Also changes the code to respect the skip-user-settings command-line flag.

Bug report: https://github.com/openrails/openrails/pull/306#issuecomment-803592877

### CODE REVIEW

File: Source/ORTS.Common/SettingsBase.cs [hunk 1]
```diff
@@ -43,7 +43,15 @@ protected enum Source
         /// <summary>The store of the settings</summary>
         protected SettingsStore SettingStore { get; private set; }
         /// <summary>Translates name of a setting to its source</summary>
-        protected readonly Dictionary<string, Source> Sources = new Dictionary<string, Source>();
+        protected readonly IDictionary<string, Source> Sources = new Dictionary<string, Source>();
+        /// <summary>
+        /// Translates name of a setting to its value when first loaded.
+        /// </summary>
+        protected readonly IDictionary<string, object> InitialValues = new Dictionary<string, object>();
```
Comment (twpol) [inline Source/ORTS.Common/SettingsBase.cs:50]: The save system (`SettingsBase.Save` IIRC) already handles the prospect of saving a setting with a value equal to its default (it removes it), so this `InitialValues` might be redundant?

File: Source/RunActivity/Program.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ static void Main(string[] args)
             game.PushState(new GameStateRunActivity(args));
             game.Run();
 
-            settings.Save();
+            settings.SaveInGameChanges();
```
Comment (twpol) [inline Source/RunActivity/Program.cs:68]: An objection to this whole approach is that saving on exit requires the program to exit cleanly, including the computer not crashing.

And targeting these specific settings is contrary to what the existing code does, which is explicitly saving when things are changed: https://github.com/openrails/openrails/blob/69600d7daa88e3340bebe894aaa0f226b81d3495/Source/RunActivity/Viewer3D/Popups/Window.cs#L105 (granted that also doesn't respect `/skip-user-settings` but that would be an easy fix too)

While this PR improves the behaviour of the newly added saving of settings, it is still disjoint from the pre-existing method and I would like to see that addressed somewhere.

**Comment by YoRyan** [comment]
Thanks for your feedback, James. I've adopted the save-immediately-upon-change approach you suggested. To accomplish this, I created a new class; I do wish I had a niftier way, but the C# typing system makes it difficult to get fancy.

**Review by twpol — APPROVED** [review]
Thanks very much, this is great!

**Comment by YoRyan** [comment]
Thanks! I'll give it a week in unstable.

**Pull request merged**